lib/src/block_syntaxes/footnote_def_syntax.dart

```diff
@@ -0,0 +1,74 @@
+import '../ast.dart' show Element, Node;
+import '../block_parser.dart' show BlockParser;
+import '../line.dart';
+import '../patterns.dart' show footnotePattern;
+import 'block_syntax.dart' show BlockSyntax;
+import 'empty_block_syntax.dart';
+import 'paragraph_syntax.dart';
+import 'setext_header_syntax.dart';
+
+class FootnoteDefSyntax extends BlockSyntax {
+  const FootnoteDefSyntax();
+
+  @override
+  RegExp get pattern => footnotePattern;
+
+  @override
+  Node? parse(BlockParser parser) {
+    final current = parser.current.content;
+    final match = pattern.firstMatch(current)!;
+    final label = match[2]!;
+    final refs = parser.document.footnoteReferences;
+    refs[label] = 0;
+
+    final id = Uri.encodeComponent(label);
+    parser.advance();
+    final lines = [
+      Line(current.substring(match[0]!.length)),
+      ...parseChildLines(parser),
+    ];
+    // Linebreak in secondary paragraph did not match LineBreakSyntax
+    // so there is no `<br />`, unlike github.
+    final children = BlockParser(lines, parser.document).parseLines();
+    return Element('li', children)
+      ..attributes['id'] = 'fn-$id'
+      ..attributes['_label_'] = label;
```

Comment:
Instead of a fake HTML attribute which is removed later, we can add a field to Element called `footnoteLabel` or something similar. Like how Element has a `generatedId` for header IDs.

Reply:
a new field make `Element` inflating, and it is a field hardly could be used by other subtype. What about using 'data attribute' from [HTML syntax](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes), attributes to store some extra information, just like `data-footnote-label`?

Reply:
I don't think we have any memory inflation concerns for this package. It doesn't outweigh the importance of the readability of the code here. A field brings in type support, typo prevention, other forms of static analysis, possible runtime optimizations, etc.